src/cbmc/cbmc_parse_options.cpp

```diff
@@ -685,9 +701,36 @@ int cbmc_parse_optionst::get_goto_program(
     {
       status() << "Reading GOTO program from file " << eom;
 
-      if(read_object_and_link(*it, symbol_table, goto_functions,
-                              get_message_handler()))
+      if(read_object_and_link(
+        *it,
+        symbol_table,
+        goto_functions,
+        get_message_handler()))
+      {
         return 6;
+      }
+    }
+
+    if(cmdline.isset("function"))
+    {
+      const std::string &function_id=cmdline.get_value("function");
+      const auto &desired_entry_function=
+        symbol_table.symbols.find(function_id);
+
+      if(desired_entry_function!=symbol_table.symbols.end())
+      {
+        languaget *language=get_language_from_mode(
+          desired_entry_function->second.mode);
+        language->regenerate_start_function(
```

Comment:
I have swapped to using `unique_ptr` as that seems to communicate intent clearer. However, if is potential memory leak otherwise, I think `get_language_from_mode` has a misleading name (get to me certainly doesn't suggest creating a new object. This code is now in [`rebuild_goto_start_functiont::operator()`](https://github.com/diffblue/cbmc/pull/338/files#diff-d0be7107cb596f16355a14b43aa0d521R72)